compute_sdk/globus_compute_sdk/sdk/web_client.py

```diff
@@ -62,6 +63,8 @@ def __init__(
                 )
             function_name = function.__name__
             function_code = _get_packed_code(function, serializer=serializer)
+            if description is None:
+                description = inspect.getdoc(function)
 
```

Comment:
I mentioned this in the first comment, but I strongly consider mentioning this detail or interaction in the `Client.register_function`.  It would not be uncouth to give code examples of both `description="explicit doc"` and `"""implicit docstring"""`

(I recognize that this is perhaps slightly more than you bargained for with this PR, but the current documentation of that function is ... weak.)